### PR TITLE
r-42 wip

### PR DESCRIPTION
**Pull request by bezirg**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by michaelpj** [comment]
@jbgi the devshell collision issue strikes again:
```
error: builder for '/nix/store/7jxmsvmsg4hpnd4alkhjic3j86mkkpd8-devshell-dir.drv' failed with exit code 25;
       last 1 log lines:
       > error: collision between `/nix/store/bbhhfphcg1adlpcmjcmcbps3rpmqr62v-hsc2hs-exe-hsc2hs-0.68.8/bin/hsc2hs' and `/nix/store/yi40pq5df7w6mp3m3081wjx9b2ndn1wq-ghc-shell-for-packages-ghc-9.2.4-env/bin/hsc2hs'
       For full logs, run 'nix log /nix/store/7jxmsvmsg4hpnd4alkhjic3j86mkkpd8-devshell-dir.drv'.
error: 1 dependencies of derivation '/nix/store/d7a5ddsc8g5hcij5k6lbbf8m36mvdrvy-plutus-shell-env.drv' failed to build
```

unsure if we even have a workaround for this?

**Comment by jbgi** [comment]
@michaelpj fix in https://github.com/input-output-hk/haskell.nix/pull/1769

File: doc/read-the-docs-site/plutus-doc.cabal [hunk 1]
```diff
@@ -69,7 +69,7 @@ executable doc-doctests
     , base               >=4.9      && <5
     , bytestring
     , containers
-    , flat
+    , flat <0.6
```
Comment (michaelpj) [inline doc/read-the-docs-site/plutus-doc.cabal:72]: <0.5 ?

Comment (michaelpj) [reply doc/read-the-docs-site/plutus-doc.cabal:72]: We can _probably_ just pin this in `plutus-core`, since everything else depends on that. Not sure.

**Review by michaelpj — COMMENTED** [review]
We should lower-bound `inline-r` also.

**Comment by bezirg** [comment]
Closed in favor of #4970

**Pull request closed without merging**